combat-trainer.lic

```diff
@@ -566,7 +566,11 @@ class SpellProcess
     echo("  @offensive_spell_cycle: #{@offensive_spell_cycle}") if $debug_mode_ct
 
     @spell_data = get_data('spells').spell_data
-
+    @offensive_spells.each.with_index do |spell, i| 
+      if @spell_data[spell['name']]
```

Comment:
I think it would be more idiomatic to do something like:
```
@offensive_spells
  .select { |spell| @spell_data[spell['name']] }
  .each.with_index do |spell, i| 
```